darts/models/torch_forecasting_model.py

```diff
@@ -71,26 +71,28 @@ def _cat_with_optional(tsr1: torch.Tensor, tsr2: Optional[torch.Tensor]):
     def __len__(self):
         return len(self.ts_dataset)
 
-    def __getitem__(self, idx: int) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+    def __getitem__(self, idx: int) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
         """
         Cast the content of the dataset to torch tensors
         """
         item = self.ts_dataset[idx]
 
-        if len(item) == 2:
+        if isinstance(self.ts_dataset, SimpleInferenceDataset):
```

Comment:
maybe you could check that it's an `TimeSeriesInferenceDataset` instead? (slightly more general)